tasks/main.yml

```diff
@@ -1,24 +1,49 @@
 ---
-- name: Get tarfile
-  command: wget https://github.com/kr/beanstalkd/archive/v1.9.tar.gz -q -O /tmp/beanstalkd.tar.gz creates=/tmp/beanstalkd.tar.gz
+- name: Discover if there is already a version of beanstalkd installed.
+  shell: which beanstalkd > /dev/null && beanstalkd -v | cut -d' ' -f2
+  changed_when: False
+  failed_when: False
+  register: installed_beanstalkd_version
 
-- name: Unarchive
-  command: tar -zxvf /tmp/beanstalkd.tar.gz chdir=/tmp creates=/tmp/beanstalkd-1.9
+- name: Ensure beanstalkd user exists.
+  user: name={{ beanstalkd_user }} comment="Beanstalkd Server" createhome=no shell=/bin/false system=yes state=present
+  register: user_changed
 
-- name: Make
-  command: make chdir=/tmp/beanstalkd-1.9/ creates=/tmp/beanstalkd-1.9/beanstalkd
+- name: Ensure persistence storage location exists and has proper ownership.
+  file: path="{{ beanstalkd_storage }}" state=directory owner={{ beanstalkd_user }} group={{ beanstalkd_user }} recurse=true
+  register: storage_dir_changed
+  notify: restart beanstalkd
 
-- name: Install
-  command: make install chdir=/tmp/beanstalkd-1.9/ creates=/usr/local/bin/beanstalkd
+- name: Get tarfile.
+  command: wget https://github.com/kr/beanstalkd/archive/v{{ beanstalkd_version }}.tar.gz -q -O /tmp/beanstalkd-{{ beanstalkd_version }}.tar.gz creates=/tmp/beanstalkd-{{ beanstalkd_version }}.tar.gz
+  when: installed_beanstalkd_version.stdout != beanstalkd_version
 
-- name: Copy beanstalkd startup script
-  copy: src=beanstalkd.initd dest=/etc/init.d/beanstalkd mode=0755
+- name: Unarchive.
+  command: tar -zxvf /tmp/beanstalkd-{{ beanstalkd_version }}.tar.gz chdir=/tmp creates=/tmp/beanstalkd-{{ beanstalkd_version }}
+  when: installed_beanstalkd_version.stdout != beanstalkd_version
 
-- name: Copy config
-  template: src=beanstalkd dest=/etc/default/beanstalkd
+- name: Make.
+  command: make chdir=/tmp/beanstalkd-{{ beanstalkd_version}}/ creates=/tmp/beanstalkd-{{ beanstalkd_version}}/beanstalkd
+  when: installed_beanstalkd_version.stdout != beanstalkd_version
```

Comment:
Did I add this 'when' ? I'd think the 'creates' handles this situation, and the when is overkill. On the other hand, the when skips the task, which should be quicker than the file-existance check... (repeated)